### PR TITLE
Do_Undo 'forceUpdateClientUndoList' method option

### PR DESCRIPTION
**Pull request by ravo10**
I found that in some cases I need to update on the client side as well (the user undo list). This change allows that. It is tested multiple times, and I found no bugs.

### CODE REVIEW

File: garrysmod/lua/includes/modules/undo.lua [hunk 1]
```diff
@@ -1,451 +1,486 @@
...
+						end
+	
+					end
+	
+				end
+			end
+
+		end
+	end
+
```
Comment (ravo10) [inline garrysmod/lua/includes/modules/undo.lua:389]: The logic

File: garrysmod/lua/includes/modules/undo.lua [hunk 2]
```diff
@@ -1,451 +1,486 @@
...
+	Current_Undo = nil
+
+end
+
+--[[---------------------------------------------------------
+	Undos an undo
+-----------------------------------------------------------]]
+function Do_Undo( undo, forceUpdateClientUndoList )
```
Comment (ravo10) [inline garrysmod/lua/includes/modules/undo.lua:339]: The parameter

File: garrysmod/lua/includes/modules/undo.lua [hunk 3]
```diff
@@ -1,451 +1,486 @@
...
+			func[ 1 ]( undo, unpack( func[ 2 ] ) )
+			count = count + 1
+
+		end
+	end
+
+	-- Force-update the client list
+	-- In some instances, this is usefull. When you also want the
+	-- client to get their list updated for whatever reason.
+	if ( undo.Entities && forceUpdateClientUndoList ) then
```
Comment (ravo10) [inline garrysmod/lua/includes/modules/undo.lua:358]: The method

**Comment by ravo10** [comment]
I will try fix that weird issue I am having where git adds the whole file, but not locally..

**Comment by ravo10** [comment]
Jeesuz. I am learning

**Comment by robotboy655** [comment]
What about just using `gmod_undo` /  `gmod_undonum <num>` console commands instead?

**Comment by ravo10** [comment]
> What about just using `gmod_undo` / `gmod_undonum <num>` console commands instead?

Yes, that works also! I am just tried making a lazy way.

**Comment by robotboy655** [comment]
Closing this as well then, using the console commands is a better solution. I have also added this info to the wiki.

**Comment by ravo10** [comment]
> Closing this as well then, using the console commands is a better solution. I have also added this info to the wiki.

I don't use GitHub much. Yes!

**Pull request closed without merging**